spec/faraday/tracer_spec.rb

```diff
@@ -16,11 +16,27 @@
     expect(span.tags['span.kind']).to eq('client')
   end
 
+  describe 'parent_span' do
+    it 'allows to pass a pre-created parent span' do
+      parent_span = tracer.start_span("parent_span")
+      expect(tracer).to receive(:start_span).with(any_args, hash_including(child_of: parent_span)).and_call_original
+      call(method: :post, span: parent_span)
+    end
+
+    it 'allows to pass a block as a parent span provider' do
+      parent_span = tracer.start_span("parent_span")
+      parent_span_provider = lambda { parent_span }
+
+      call(method: :post, span: parent_span_provider)
```

Comment:
hmm `parent_id` is not defined in OT specification. Shame, I missed the expectation for the test ... ;)